### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by alexec**
Enable the default lint checks (also fix those lint errors).

### CODE REVIEW

File: pkg/apiclient/grpcproxy.go [hunk 1]
```diff
@@ -128,10 +128,8 @@ func (c *client) startGRPCProxy() (*grpc.Server, net.Listener, error) {
 			}
 
 			go func() {
-				select {
-				case <-stream.Context().Done():
-					util.Close(resp.Body)
-				}
+				<-stream.Context().Done()
```
Comment (alexec) [inline pkg/apiclient/grpcproxy.go]: Significant change?

File: util/kustomize/kustomize.go [hunk 2]
```diff
@@ -209,7 +209,7 @@ func getImageParameters(objs []*unstructured.Unstructured) []Image {
 		images = append(images, getImages(obj.Object)...)
 	}
 	sort.Slice(images, func(i, j int) bool {
-		return j < j
```
Comment (alexec) [inline util/kustomize/kustomize.go]: This was a bug.

File: util/project/util.go [hunk 3]
```diff
@@ -174,8 +174,8 @@ func AddGroupToRole(p *v1alpha1.AppProject, roleName, group string) (bool, error
 	if err != nil {
 		return false, err
 	}
-	for _, group := range role.Groups {
-		if group == group {
+	for _, roleGroup := range role.Groups {
+		if group == roleGroup {
```
Comment (alexec) [inline util/project/util.go]: Significant change.

File: util/project/util.go [hunk 4]
```diff
@@ -190,8 +190,8 @@ func RemoveGroupFromRole(p *v1alpha1.AppProject, roleName, group string) (bool,
 	if err != nil {
 		return false, err
 	}
-	for i, group := range role.Groups {
-		if group == group {
+	for i, roleGroup := range role.Groups {
+		if group == roleGroup {
```
Comment (alexec) [inline util/project/util.go]: Significant change.

File: test/manifests_test.go [hunk 5]
```diff
@@ -19,6 +19,9 @@ func TestBuildManifests(t *testing.T) {
 	assert.True(t, Contains(out, "KustomizeVersion:2") || Contains(out, "KustomizeVersion:v2"), "kustomize should be version 2")
 
 	err = filepath.Walk("../manifests", func(path string, f os.FileInfo, err error) error {
+		if err != nil {
```
Comment (alexmt) [inline test/manifests_test.go]: This looks reduntant

File: util/db/helmrepository.go [hunk 6]
```diff
@@ -15,7 +15,9 @@ import (
 
 func getHelmRepoCredIndex(s *settings.ArgoCDSettings, repoURL string) int {
 	for i, cred := range s.HelmRepositories {
-		if strings.ToLower(cred.URL) == strings.ToLower(repoURL) {
+		// work-around the SA6005 check as I'm not sure that the suggested changes is correct
```
Comment (alexmt) [inline util/db/helmrepository.go]: Found description of SA6005 https://staticcheck.io/docs/checks#SA6005. It suggests using `strings.EqualFold(s1, s2).`

File: util/kustomize/kustomize.go (hunk 2 above)
Comment (alexmt) [reply util/kustomize/kustomize.go]: 👍

File: util/project/util.go (hunk 3 above)
Comment (alexmt) [reply util/project/util.go]: 👀

**Review by alexmt — CHANGES_REQUESTED** [review]
Nice improvement! Added one comment about SA6005

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**